sdk/core/Azure.Core/src/TokenRequestContextOptions.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Options for <see cref="TokenRequestContext"/> initialization.
+    /// </summary>
+    public struct TokenRequestContextOptions
```

Comment:
This was added to avoid ctor ambiguity errors that come with an additional optional string parameter in a new overload. This will make adding additional properties easier in the future as well.